### ISSUE TITLE
Add lint inspection for unused value

### ISSUE THREAD

**Issue by KotlinIsland** [denoland/deno_lint#637]
```ts
(1).toFixed;
1;
```
These should be reported as unused values.

**Comment by rivy** [comment]
This could catch a lot of otherwise silent errors when using common JS/TS constructions.

It looks like `tslint` and SonarLint have implemented a version of this rule (see <https://github.com/typescript-eslint/typescript-eslint/issues/1314> and [SonarJS rule ~ no-ignored-return](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/docs/rules/no-ignored-return.md)).

The unicorn plugin for eslint is has an accepted an [issue requesting an implementation](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/741) for this rule which discusses some relevant issues.
